crypto/rand/md_rand.c

```diff
@@ -555,6 +555,18 @@ int ssleay_rand_bytes(unsigned char *buf, int num, int pseudo, int lock)
     return (0);
 }
 
+/*
+ * Returns ssleay_rand_bytes(), enforcing a reseeding from the
+ * system entropy sources using RAND_poll() before generating
+`* the random bytes.
+ */
+
+int ssleay_rand_bytes_from_system(unsigned char *buf, int num)
+{
+    initialized = 0;
+    return ssleay_rand_bytes(buf, num, 0, 0);
```

Comment:
I assume that the locking that allows us to pass ```lock=0``` to ```ssleay_rand_bytes()``` also serves to serialize the accesses to ```initialized``` (that is, someone calling this function already holds the ```CRYPTO_RAND_LOCK```).

Reply:
That's my assumption, too, but I will double-check it in the debugger using my test program.